src/server/req_message.c

```diff
@@ -341,7 +342,18 @@ req_relnodesjob(struct batch_request *preq)
 	if ((nodeslist != NULL) && (nodeslist[0] == '\0')) {
 		nodeslist = NULL;
 	}
-	rc = free_sister_vnodes(pjob, nodeslist, msg, LOG_BUF_SIZE, preq);
+
+	if (preq->rq_extend && *preq->rq_extend) {
+		keep_select = strchr(preq->rq_extend, '=');
+		if (!keep_select || !*(keep_select+1))
+			rc = PBSE_INVALSELECTRESC;
+		if (rc || (rc = validate_perm_res_in_select(++keep_select, 1))) {
```

Comment:
Just keep in mind of some subtlety here. The validate_perm_res_in_select() here would strdup() a value for resc_in_err, and because it only happens with a return code of PBSE_UNKRESC, the req_reject() would cause calling set_err_msg(), which in turn would free(resc_in_err).